pkg/app/sensor/app.go

```diff
@@ -527,6 +527,43 @@ func readCommandFile() (command.StartMonitor, error) {
 		return cmd, fmt.Errorf("could not decode command: %w", err)
 	}
 
+	// Dockerfile allows to define ENTRYPOINT[] + CMD[]
+	// while sensor's StartMonitor command historically
+	// allows providing only AppName + AppArgs[] making
+	// the sensor's approach inherently inferior.
+	// In the standalone mode the CMD[] part can be
+	// overridden at the run-time, but to be able to
+	// safely override the StartMonitor command, we have
+	// to differentiate between the constant ENTRYPOINT[]
+	// and the dynamic CMD[] part. Because of that, an
+	// extra field StartMonitor.Entrypoint was added.
+	// However, to avoid cascading changes to the rest of
+	// the sensor's code, we do a conversion back to
+	// the AppName + AppArgs[] format early on.
+
+	// First, check if there is a run-time override of the CMD[] part.
+	if args := flag.Args(); len(args) > 0 {
```

Comment:
possible to have overlapping args... which might cause unexpected side effects (we should review those)

Reply:
IIRC, `Args` doesn't include flags. So, if we stick with flags only for the sensor itself, we'll probably be fine. Unless I'm missing something, of course.

Here is how the agreed `ENTRYPOINT` needs to look like (on the instrumentor side): `ENTRYPOINT [ "/path/to/sensor", "-m=standalone", "-c=/path/to/commands.json", "--" ]`. I.e., everything from the `CMD` (if provided at runtime) will be appended after `--` which is a special separator value (supported by the `flag` package). I.e., if a user will pass another `-m` it won't become a flag, but remain an (unparsed) arg.

Reply:
The `--` trick is nice. The other option is to have a default config/commands file location, so we don't need to have explicit params